rotkehlchen/rotkehlchen.py

```diff
@@ -168,6 +169,29 @@ def __init__(self, args: argparse.Namespace) -> None:
         self.shutdown_event = gevent.event.Event()
         self.migration_manager = DataMigrationManager(self)
 
+    def kill_running_tx_query_tasks(
+            self,
+            blockchain: SupportedBlockchain,
+            accounts: ListOfBlockchainAddresses,
+    ) -> None:
+        """Checks for running greenlets related to transactions query and kills them."""
+        assert self.task_manager is not None
+        log.debug(
+            f'Attempting to kill running transaction queries tasks '
+            f'from api task greenlets {self.api_task_greenlets} and '
+            f'task manager running greenlets {self.task_manager.running_greenlets}',
+        )
+        for task in self.api_task_greenlets:
+            if task.is_txn_query is True:
+                task.greenlet.kill()
+                self.api_task_greenlets.remove(task)
+
+        task_names = {f'Query {blockchain.name.lower()} transactions for {account}' for account in accounts}  # noqa: E501
```

Comment:
Hmmm, this comparison by name looks brittle to me. Maybe compare with a function or add a special flag for such greenlets (like `is_txn_query`) or something like that?